src/ts/ssh/pipeExtensions.ts

```diff
@@ -168,8 +177,12 @@ export class PipeExtensions {
 		toChannel: SshChannel,
 		data: Buffer,
 	): Promise<void> {
-		await toChannel.send(data, CancellationToken.None);
-		channel.adjustWindow(data.length);
+		// Seems that somehow data gets corrupted/disposed? so do a copy first thing
```

Comment:
The data buffer is re-used when receiving the next message. In many cases a channel data handler can process the data synchronously without any need for allocating and copying the buffer.

But below, the `send()` method is async because it may need to wait for the channel window to open up before sending the full buffer. So, making a copy of the buffer at this point is the safe thing to do.

It _might_ be possible to optimize such that the copy can be skipped if `send()` manages to complete immediately (which should be the most common case), but I think that would be complicated.